src/GitHubApps/src/DarcBot/CryptoHelper.cs

```diff
@@ -0,0 +1,129 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.Security.Cryptography;
+
+static class CryptoHelper
+{
+    public static RSAParameters GetRsaParameters(string privateKeyBase64)
```

Comment:
??? Do we need to parse the private key? I don't think any other instances of our interactions with github do this.  This is a lot of very complicated code that'd I'd rather avoid owning.

Reply:
Is there a sample of somewhere else that we do this?  This was the way I found to make authentication work

> This JWT needs to be signed with the RS256 algorithm (RSA signature with SHA256); in order to sign it, you need a private key, which must be generated from the GitHub app settings page. You can download the private key in PEM format, and store it somewhere your app can access it. Unfortunately, the .NET APIs to generate and sign a JWT don’t handle the PEM format, they need an RSAParameters object… 

If there's pre-existing work where we do this in a simpler manner, I'm all for it (it was a bit of a pain to sort out a reasonable way to do this).  The current method pulls the PEM key from Azure Key Vault and does the parsing.

Reply:
The "GitHubAppTokenProvider" and "InMemoryCacheInstallationLookup" classes should have all the logic we need.  The only bits that it needs are some options pointing to the various secrets.  We DI them in the other services, but you should be able to create them as static members in this class without too much work.

We probably need to pull the "InMemoryCacheInstallationLookup" class out into Shared/Microsoft.DotNet.GitHub.Authentication, so that you can reference it, but that should do the trick.

Reply:
Super happy we already had some of this authentication stuff using the GitHubJwtFactory package.  Thanks for the pointer.